### PR TITLE
added non-core-qn workflow

### LINKED ISSUE #8847: Automate Labeling of Issues Based on Comments from Non-Core Members
**Is your feature request related to a problem? Please describe.**
Currently, there is no way to track issues with unresolved comments. Contributors, interns, or anyone other than core team members may have shared a comment that requires action from the core team, but it might be missed, causing the issue to stall. 

**Describe the solution you'd like**
To identify such cases, a label can be automatically added to the issue whenever someone other than a core team member comments.

Feature Requirements:
- Automatically add a "question" label to an issue whenever a non-core team member comments.
- Once a core team member replies or comments, the "question" label should be automatically removed.

### PR DESCRIPTION
**Pull request by tellmeY18**
## Proposed Changes

- Fixes #8847 
- added a new workflow file .github/workflows/notify-non-core-qn.yml
- Change 2
- More?

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Introduced an automated notification system for the core team when non-core contributors post questions in issue comments.
	- The system labels relevant issues and sends notifications to the core team via Slack, enhancing communication and response efficiency.

### CODE REVIEW

File: .github/workflows/notify-non-core-qn.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+              role: 'all'
+            });
+            const isOrgMember = orgMembers.some(member => member.login === commenter);
+            if (!isOrgMember) {
+              return commenter; // Proceed if not an organization member
+            }
+      - name: Check if comment is a question
+        id: check_question
+        run: |
+          echo "${{ github.event.comment.body }}" | grep -Eq "(\?|what|how|why|when|who)"
```
[the remark this answers is not in the document]
Comment (rithviknishad) [reply .github/workflows/notify-non-core-qn.yml]: @tellmeY18 

we could check if the comment is a question in the javascript run script step in the previous step instead right?

File: .github/workflows/notify-non-core-qn.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            await github.issues.addLabels({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: context.payload.issue.number,
+              labels: ['question'],
+            });
```
Comment (rithviknishad) [inline .github/workflows/notify-non-core-qn.yml]: we could merge these three github-script step as 1 step right?

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: .github/workflows/notify-non-core-qn.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tellmeY18) [reply .github/workflows/notify-non-core-qn.yml]: updated in the commit below

File: .github/workflows/notify-non-core-qn.yml (hunk 2 above)
Comment (tellmeY18) [reply .github/workflows/notify-non-core-qn.yml]: updated to make it single step

File: .github/workflows/notify-non-core-qn.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+          script: |
+            const commenter = context.payload.comment.user.login;
+            const { data: orgMembers } = await github.orgs.listMembers({
+              org: context.repo.owner,
+              role: 'all'
+            });
+            const isOrgMember = orgMembers.some(member => member.login === commenter);
+            if (!isOrgMember) {
+              const commentBody = context.payload.comment.body.toLowerCase();
+              const questionKeywords = ["?", "what", "how", "why", "when", "who"];
```
Comment (bodhish) [inline .github/workflows/notify-non-core-qn.yml]: This should be a config that can be read from our variable in GitHub

File: .github/workflows/notify-non-core-qn.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                    repository: context.repo.repo
+                  }
+                });
+                await github.issues.addLabels({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  issue_number: context.payload.issue.number,
+                  labels: ['question'],
+                });
+                const slackWebhook = process.env.SLACK_WEBHOOK;
```
Comment (bodhish) [inline .github/workflows/notify-non-core-qn.yml]: Test if this is working

File: .github/workflows/notify-non-core-qn.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+                  client_payload: {
+                    issue_number: context.payload.issue.number,
+                    repository: context.repo.repo
+                  }
+                });
+                await github.issues.addLabels({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  issue_number: context.payload.issue.number,
+                  labels: ['question'],
```
Comment (bodhish) [inline .github/workflows/notify-non-core-qn.yml]: The labels to add should also be part of the variable. https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables

File: .github/workflows/notify-non-core-qn.yml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const commenter = context.payload.comment.user.login;
+            const { data: orgMembers } = await github.orgs.listMembers({
+              org: context.repo.owner,
+              role: 'all'
+            });
+            const isOrgMember = orgMembers.some(member => member.login === commenter);
```
Comment (bodhish) [inline .github/workflows/notify-non-core-qn.yml]: This should be an array of usernames that we set as a variable.

File: .github/workflows/notify-non-core-qn.yml (hunk 4 above)
Comment (bodhish) [reply .github/workflows/notify-non-core-qn.yml]: also this should go into a specific channel

**Review by bodhish — CHANGES_REQUESTED** [review]

**Comment by bodhish** [comment]
@tellmeY18 status 🤔

**Comment by bodhish** [comment]
@coderabbitai full review

**Comment by bodhish** [comment]
@coderabbitai full review

File: .github/workflows/notify-non-core-qn.yml (hunk 3 above)
Comment (tellmeY18) [reply .github/workflows/notify-non-core-qn.yml]: resolved.

File: .github/workflows/notify-non-core-qn.yml (hunk 4 above)
Comment (tellmeY18) [reply .github/workflows/notify-non-core-qn.yml]: tested and we can add the channel webhook as a secret.

File: .github/workflows/notify-non-core-qn.yml (hunk 5 above)
Comment (tellmeY18) [reply .github/workflows/notify-non-core-qn.yml]: added as a variable that can be added in the gh variables section.

File: .github/workflows/notify-non-core-qn.yml (hunk 6 above)
Comment (tellmeY18) [reply .github/workflows/notify-non-core-qn.yml]: updated.

File: .github/workflows/notify-non-core-qn.yml [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+                    },
+                    body: JSON.stringify(message)
+                  });
+                  console.log('Slack notification sent successfully');
+                } catch (error) {
+                  console.error('Error sending Slack notification:', error.message);
+                }
+              }
+            }
+            console.log('Script ended');
```
[the remark this answers is not in the document]
Comment (tellmeY18) [reply .github/workflows/notify-non-core-qn.yml]: resolved.

File: .github/workflows/notify-non-core-qn.yml [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+            console.log('Script started');
+            const commenter = context.payload.comment.user.login;
+            console.log('Commenter:', commenter);
+            const coreUsernames = process.env.ALLOWED_USERNAMES.split(',');
+            const isOrgMember = coreUsernames.includes(commenter);
+            console.log('Is commenter an org member?', isOrgMember);
+
+            if (!isOrgMember) {
+              const commentBody = context.payload.comment.body.toLowerCase();
+              console.log('Comment body:', commentBody);
```
[the remark this answers is not in the document]
Comment (tellmeY18) [reply .github/workflows/notify-non-core-qn.yml]: resolved.

File: .github/workflows/notify-non-core-qn.yml [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+                    method: 'POST',
+                    headers: {
+                      'Content-Type': 'application/json'
+                    },
+                    body: JSON.stringify(message)
+                  });
+                  console.log('Slack notification sent successfully');
+                } catch (error) {
+                  console.error('Error sending Slack notification:', error.message);
+                }
```
[the remark this answers is not in the document]
Comment (tellmeY18) [reply .github/workflows/notify-non-core-qn.yml]: resolved

**Review by rithviknishad — APPROVED** [review]

**Review by bodhish — APPROVED** [review]

File: .github/workflows/notify-non-core-qn.yml [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+              console.log('Comment body:', sanitizedComment);
+              if (containsQuestionKeywords(sanitizedComment, process.env.QUESTION_KEYWORDS)) {
+                try {
+                  console.log('Adding labels to the issue');
+                  await addLabelsToIssue(github, context, process.env.QUESTION_LABELS);
+                  console.log('Labels added successfully');
+                  const issueUrl = `https://github.com/${context.repo.owner}/${context.repo.repo}/issues/${context.payload.issue.number}`;
+                  const issueTitle = context.payload.issue.title;
+                  const issueNumber = context.payload.issue.number;
+                  console.log('Issue URL:', issueUrl);
```
Comment (bodhish) [inline .github/workflows/notify-non-core-qn.yml:69]: I don't think we should keep these logs

File: .github/workflows/notify-non-core-qn.yml [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+  issues: write
+  pull-requests: write
+jobs:
+  notify_core_team:
+    runs-on: ubuntu-latest
+    env:
+      ALLOWED_USERNAMES: ${{ vars.ALLOWED_USERNAMES }}
+      QUESTION_KEYWORDS: ${{ vars.QUESTION_KEYWORDS }}
+      QUESTION_LABELS: ${{ vars.QUESTION_LABELS }}
+      SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
```
Comment (bodhish) [inline .github/workflows/notify-non-core-qn.yml:15]: The steps should be skipped if the variables/ secrets are not available

**Review by bodhish — CHANGES_REQUESTED** [review]

**Pull request merged**